src/Microsoft.Health.Fhir.TemplateManagement/Overlay/OverlayFileSystem.cs

```diff
@@ -0,0 +1,139 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.IO;
+using EnsureThat;
+using Microsoft.Health.Fhir.TemplateManagement.Models;
+
+namespace Microsoft.Health.Fhir.TemplateManagement.Overlay
+{
+    public class OverlayFileSystem : IOverlayFileSystem
+    {
+
+        public OverlayFileSystem(string workingFolder)
+        {
+            EnsureArg.IsNotNull(workingFolder, nameof(workingFolder));
+
+            WorkingFolder = workingFolder;
+            WorkingImageLayerFolder = Path.Combine(workingFolder, Constants.HiddenLayersFolder);
+            WorkingBaseLayerFolder = Path.Combine(workingFolder, Constants.HiddenBaseLayerFolder);
+        }
+
+        public string WorkingFolder { get; set; }
+
+        public string WorkingImageLayerFolder { get; set; }
+
+        public string WorkingBaseLayerFolder { get; set; }
+
+        public OCIFileLayer ReadMergedOCIFileLayer()
+        {
+            var filePaths = Directory.EnumerateFiles(WorkingFolder, "*.*", SearchOption.AllDirectories);
+            Dictionary<string, byte[]> fileContent = new Dictionary<string, byte[]>() { };
+            foreach (var oneFile in filePaths)
+            {
+                if (!Path.GetRelativePath(WorkingFolder, oneFile).Contains(Constants.HiddenImageFolder))
+                {
+                    fileContent.Add(Path.GetRelativePath(WorkingFolder, oneFile), File.ReadAllBytes(oneFile));
+                }
+            }
+
+            OCIFileLayer fileLayer = new OCIFileLayer() { FileContent = fileContent };
+            return fileLayer;
+        }
+
+        public void WriteMergedOCIFileLayer(OCIFileLayer oneLayer)
+        {
+            EnsureArg.IsNotNull(oneLayer, nameof(oneLayer));
+
+            Directory.CreateDirectory(WorkingFolder);
+            foreach (var oneFile in oneLayer.FileContent)
+            {
+                var directory = Path.GetDirectoryName(Path.Combine(WorkingFolder, oneFile.Key));
```

Comment:
We can add a new variable ```var filePath = Path.Combine(WorkingFolder, oneFile.Key).```
It looks confusing that we don't use directory in line 56

Reply:
This is just for readibility